service-spec/schema-validator/src/main/java/io/hyscale/schema/validator/JsonSchemaValidator.java

```diff
@@ -23,36 +23,47 @@
 import com.github.fge.jsonschema.main.JsonSchemaFactory;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.stereotype.Component;
 
 import java.io.IOException;
 
 /**
  * Created by vijays on 18/9/19.
+ * Compares input String,JsonNode with the reference Json Schema passed and returns report.
  */
-@Component
-public class ServiceSpecValidator {
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceSpecValidator.class);
-    private final JsonSchemaFactory factory = JsonSchemaFactory.byDefault();
-
-    private static final String SERVICE_SCHEMA_FILE = "/v1.0.0-standalone/service-spec.json";
-
-    public ProcessingReport validate(String inputServiceSpec) throws IOException, ProcessingException {
-
-        final JsonNode serviceSpecSchemaJsonNode = JsonLoader.fromResource(SERVICE_SCHEMA_FILE);
-        final JsonSchema schema = factory.getJsonSchema(serviceSpecSchemaJsonNode);
+public class JsonSchemaValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JsonSchemaValidator.class);
+    private static final JsonSchemaFactory factory = JsonSchemaFactory.byDefault();
 
+    /**
+     * Validates whether the given input string satisfies schema.
+     *
+     * @param inputServiceSpec String input
+     * @param schema Reference Json Schema
+     * @return ProcessingReport
+     * @throws IOException
+     * @throws ProcessingException
+     */
+    public static ProcessingReport validate(String inputServiceSpec, String schema) throws IOException, ProcessingException {
         final JsonNode inputServiceSpecJsonNode = JsonLoader.fromString(inputServiceSpec);
-        return schema.validate(inputServiceSpecJsonNode);
+        return validateSpec(schema, inputServiceSpecJsonNode);
     }
 
-    public ProcessingReport validate(JsonNode inputServiceSchema) throws IOException, ProcessingException {
+    /**
+     * Validates whether the given input json node satisfies schema.
+     *
+     * @param inputServiceSpec JsonNode input
+     * @param schema Reference Json Schema
+     * @return ProcessingReport
+     * @throws IOException
+     * @throws ProcessingException
+     */
+    public static ProcessingReport validate(JsonNode inputServiceSpec, String schema) throws IOException, ProcessingException {
+        return validateSpec(schema, inputServiceSpec);
+    }
```

Comment:
Schema can also be json node , why is it a string ?
it should be something like validate(String schema,String jsonStr) & validate(JsonNode schemaNode, JsonNode jsonNode)